docs/data/date-pickers/localization/localization.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Pickers - Localization
+---
+
+# Pickers - Localization
+
+<p class="description">Pickers allow to support users from different locales, with formatting, RTL, and localized strings.</p>
+
+The default locale of MUI is English (United States). If you want to use other locales, follow the instructions below.
+
+## Localization
+
+Localization can impact pickers components rendering in two distincts ways: The date format, and the components attributes such as `aria-label`.
+
+### Date-engine locale
+
+Use `LocalizationProvider` to change the date-engine locale that is used to render pickers. Here is an example of changing the locale for the `date-fns` adapter:
+
+{{"demo": "LocalizedDatePicker.js"}}
+
+### Translation keys
+
+As the rest of MUI components, you can modify text and translations.
+You can find all the translation keys supported in [the source](https://github.com/mui/mui-x/blob/HEAD/packages/grid/x-date-pickers/src/locales/utils/pickersLocaleTextApi.ts)
+in the GitHub repository.
+
+You can set the locale text by using the theme provider.
+
+```jsx
+import { createTheme, ThemeProvider } from '@mui/material/styles';
+import { CalendarPicker, LocalizationProvider, bgBG } from '@mui/x-date-pickers';
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
+
+const theme = createTheme(
+  {
+    palette: {
+      primary: { main: '#1976d2' },
+    },
+  },
+  bgBG,
+);
+
+<ThemeProvider theme={theme}>
+  <LocalizationProvider dateAdapter={AdapterDateFns}>
+    <CalendarPicker />
+  </LocalizationProvider>
+</ThemeProvider>;
+```
+
+Note that `createTheme` accepts any number of arguments.
+If you are already using the [translations of the core components](/material-ui/guides/localization/#locale-text) or the [translations of the data grid](/x/react-data-grid/localization/#locale-text), you can add `bgBG` as a new argument.
+The same import works for `DataGridPro` as it's an extension of `DataGrid`.
+
+```jsx
+import { createTheme, ThemeProvider } from '@mui/material/styles';
+import { DataGrid, bgBG as dataGridBgBG } from '@mui/x-data-grid';
+import { bgBG as coreBgBG } from '@mui/material/locale';
+
+import { CalendarPicker, LocalizationProvider, bgBG } from '@mui/x-date-pickers';
+import { AdapterDateFns } from '@mui/x-date-pickers/AdapterDateFns';
+
+const theme = createTheme(
+  {
+    palette: {
+      primary: { main: '#1976d2' },
+    },
+  },
+  bgBG, // x-date-pickers translations
+  dataGridBgBG, // x-data-grid translations
+  coreBgBG, // core translations
```

Comment:
Mayber add the pickers translations to the sibling example in the data grid doc
https://mui.com/x/react-data-grid/localization/#locale-text

Reply:
Good idea